app/tests/components_tests/test_views.py

```diff
@@ -261,3 +263,84 @@ def test_civset_delete_view(
     assert response.status_code == 302
     assert ArchiveItem.objects.count() == 0
     assert DisplaySet.objects.count() == 0
+
+
+@pytest.mark.parametrize(
+    "base_object_factory,base_obj_lookup,object_factory,viewname,add_collaborator_attr,collaborator_visible_obj_count",
+    (
+        (
+            ReaderStudyFactory,
+            "reader_study",
+            DisplaySetFactory,
+            "reader-studies:display_sets",
+            "add_reader",
+            0,
+        ),
+        (
+            ArchiveFactory,
+            "archive",
+            ArchiveItemFactory,
+            "archives:items-list",
+            "add_uploader",
+            3,
+        ),
+        (
+            ArchiveFactory,
+            "archive",
+            ArchiveItemFactory,
+            "archives:items-list",
+            "add_user",
+            3,
+        ),
+    ),
+)
+@pytest.mark.django_db
+def test_civset_list_view_permissions(
+    client,
+    base_object_factory,
+    base_obj_lookup,
+    object_factory,
+    viewname,
+    add_collaborator_attr,
+    collaborator_visible_obj_count,
+):
+    user, editor, collaborator = UserFactory.create_batch(3)
+    base_obj = base_object_factory()
+    base_obj.add_editor(editor)
+    method = getattr(base_obj, add_collaborator_attr)
+    setattr(base_obj, add_collaborator_attr, method)
+    getattr(base_obj, add_collaborator_attr)(collaborator)
+    ob1, ob2, ob3 = object_factory.create_batch(
+        3, **{base_obj_lookup: base_obj}
+    )
+
+    response = get_view_for_user(
+        viewname=viewname,
+        client=client,
+        user=user,
+        reverse_kwargs={"slug": base_obj.slug},
+    )
+    assert response.status_code == 200
+    assert len(response.context["object_list"]) == 0
+
+    response = get_view_for_user(
+        viewname=viewname,
+        client=client,
+        user=collaborator,
+        reverse_kwargs={"slug": base_obj.slug},
+    )
+    assert response.status_code == 200
+    assert (
+        len(response.context["object_list"]) == collaborator_visible_obj_count
```

Comment:
Maybe add some civ sets for other objects to make sure that they're not visible too, then we know the filtering is hooked up correctly.